### DISCUSSION TITLE
Add feature light distribution control to LightEntity

### DISCUSSION THREAD

**Discussion by sMauldaeschle** [home-assistant/architecture#704] [Entity Models]
## Context

Some light fixtures are able to emit light in opposite directions (e.g. up and down). It is physically one light fixture but has more than one actual light sources that allow transitions and mixtures of up and down light for instance.

A big technology provider is Casambi whose various smart devices and chips get integrate in divers light fixture from various light manufacturers. Example products can be found from Occhio, e.g. [Sento](https://www.occhio.de/en/sento).
Another example with up/down lightning is [Philips Hue Ensis](https://www.philips-hue.com/en-gb/p/hue-white-and-colour-ambiance-ensis-pendant-light/8719514343467)). Here I did not find any information if they already support the transition.

Home Assistant and the LightEntity currently is not able to handle this feature, why I suggest to add it.

## Proposal

Add new feature to support light distribution control. This change adds the attribute `ATTR_DISTRIBUTION`, extends the LightEntity class with the member `_attr_distribution`, updates the schemas and adds the corresponding function `def distribution(self)`. The turn_on and toggle services are extend by another field `distribution`.

Since I did not know of the architectural discussions here, PRs already exist:
PR for the code changes: https://github.com/home-assistant/core/pull/63629
PR for the documentation update: https://github.com/home-assistant/home-assistant.io/pull/21088
Open work from my perspective: Include a conditional slider in the more-info dialogue to adjust the distribution in the same way as brightness.

I hope this is in line with the architectural thoughts for the LightEntity. Happy to hear your feedback and suggestions.
If anybody is interested in testing: The [home_assistant_casambi](https://github.com/hellqvio86/home_assistant_casambi) component and its [aiocasambi](https://github.com/hellqvio86/aiocasambi) library have been updated by me already. The feature is functional.

## Consequences

New light feature can be fully controlled from HA.

**Comment by frenck** [comment]
Which other integrations could benefit from this change?

**Reply by sMauldaeschle** [reply]
Basically every light integration that plans to support up and down fading (or horizontal fading) in a single light fixture.
If you're asking wrt to standardization character I also thought about realizing that integration specific but came to the conclusion that is a feature which should be controllable by the standard services and UI. Anyhow it is for sure not a default feature of a wide range of lights, more an advanced function of pricy lights.
I also thought about the effect_list but to my understanding this is not really an effect but a static control where effects might even come on top.
Hopefully in the future scenes can be configured more freely and will also benefit. Today and automation needs to be used to get the benefit already now.

**Reply by frenck** [reply]
> Basically every light integration that plans to support up and down fading (or horizontal fading) in a single light fixture.

I get that, the question was: Which integrations would be able to support this today.

As in, if we change our entity model, there must be more than just a single integration in order to warrant a model change.

**Reply by frenck** [reply]
The last question was/is very important to use as a base for our architectural decisions. As there is no continued conversation on this (the proposal ran stale), I'll close up this proposal for now.

Feel free to re-open when you feel like that is the right thing to do.
